### ISSUE TITLE
Magic Link: Some emails linked to WordPress.com accounts aren't recognized

### ISSUE THREAD

**Issue by rachelmcr** [wordpress-mobile/WordPress-Android#4794]
### Expected behavior

I expect that if I enter an email address linked to a WordPress.com account in the login form, I will proceed through the magic link login flow.

### Actual behavior

Some email addresses are not recognized as being linked to a WordPress.com account, so after entering the email address I am taken to the self-hosted login form.

Tapping "Log in to WordPress.com" takes me back to the previous login form (username/email only), but I still can't force a WordPress.com login — when I try to go to the next screen I am taken again to the self-hosted login form.

I noticed that this seemed to consistently happen with Gmail addresses using + addressing, e.g. `example+wordpress@gmail.com`.

### Steps to reproduce the behavior

1. Open the app (logged out) on the WordPress.com login screen
2. Enter a Gmail address with + addressing linked to a WordPress.com account
3. Tap the "Next" button

Result: The form takes you to the self-hosted login form, even though the email is linked to a WordPress.com account.

##### Tested on Nexus 9, Android 7.0, WPAndroid alpha-28

**Comment by aforcier** [comment]
I took a look at the cause of this bug, and it's happening because of unnecessary sanitization of the email before it's sent to the `is-available` API endpoint. The `+` sign in the email is converted to a space [here](https://github.com/wordpress-mobile/WordPress-Android/blob/a68f9f434a628a1f1bfccf8bf3260a1dacd587bf/libs/networking/WordPressNetworking/src/main/java/org/wordpress/android/networking/RestClientUtils.java#L362), which is later encoded into a `%20` when the email is sent to the endpoint, which returns an 'Invalid E-mail' error.

I also confirmed that this is already handled correctly by FluxC (it's working in our current `feature/fluxc-integration` branch).

We can still fix this in the meantime as it's preventing anyone with a `+` in their e-mail from logging in using it, but we'll be able to close this issue when FluxC is merged into `develop`.
